compiler/hlint-testing/src/Main.hs

```diff
@@ -0,0 +1,167 @@
+analyzeModule modu anns = do
+  (_, classify, hint) <- hlintSettings
+  print $ applyHints classify hint [createModuleEx anns modu]
+
+main :: IO ()
+main = do
+  args <- getArgs
+  case args of
+    [file] -> do
+      s <- readFile' file
+      flags <-
+        parsePragmasIntoDynFlags
+          (defaultDynFlags fakeSettings fakeLlvmConfig) file s
+      whenJust flags $ \flags ->
+         case parse file (flags `gopt_set` Opt_KeepRawTokenStream)s of
+            PFailed _ loc err ->
+              putStrLn (showSDoc flags (pprLocErrMsg (mkPlainErrMsg flags loc err)))
+            POk s m -> do
+              let (wrns, errs) = getMessages s flags
+              report flags wrns
+              report flags errs
+              when (null errs) $ analyzeModule m (harvestAnns s)
+    _ -> fail "Exactly one file argument required"
+  where
+    report flags msgs =
+      sequence_
+        [ putStrLn $ showSDoc flags msg
+        | msg <- pprErrMsgBagWithLoc msgs
+        ]
+    harvestAnns pst =
+      ( Map.fromListWith (++) $ annotations pst
+      , Map.fromList ((noSrcSpan, comment_q pst) : annotations_comments pst)
+      )
+
+parse :: String -> DynFlags -> String -> ParseResult (Located (HsModule GhcPs))
+parse filename flags str =
+  unP Parser.parseModule parseState
+  where
+    location = mkRealSrcLoc (mkFastString filename) 1 1
+    buffer = stringToStringBuffer str
+    parseState = mkPState flags buffer location
+
+parsePragmasIntoDynFlags :: DynFlags -> FilePath -> String -> IO (Maybe DynFlags)
+parsePragmasIntoDynFlags flags filepath str =
+  catchErrors $ do
+    let opts = getOptions flags (stringToStringBuffer str) filepath
+    (flags, _, _) <- parseDynamicFilePragma flags opts
+    return $ Just (foldl' xopt_set flags xExtensionsSet)
+  where
+    catchErrors :: IO (Maybe DynFlags) -> IO (Maybe DynFlags)
+    catchErrors act = handleGhcException reportErr
+                        (handleSourceError reportErr act)
+    reportErr e = do putStrLn $ "error : " ++ show e; return Nothing
+
+xExtensionsSet :: [Extension]
+xExtensionsSet =
+  [ RecordPuns, RecordWildCards, LambdaCase, TupleSections, BlockArguments, ViewPatterns
+  , NumericUnderscores
+  , DuplicateRecordFields, DisambiguateRecordFields
+  , ScopedTypeVariables, ExplicitForAll
+  , DataKinds, KindSignatures, RankNTypes, TypeApplications
+  , ConstraintKinds
+  , MultiParamTypeClasses, FlexibleInstances, GeneralizedNewtypeDeriving, TypeSynonymInstances
+  , DefaultSignatures, StandaloneDeriving, FunctionalDependencies, DeriveFunctor
+  , RebindableSyntax, OverloadedStrings
+  , Strict, StrictData
+  , MonadComprehensions
+  , PackageImports
+  , DamlSyntax
+  ]
+
+fakeSettings :: Settings
+fakeSettings = Settings
+  { sTargetPlatform=platform
+  , sPlatformConstants=platformConstants
+  , sProjectVersion=cProjectVersion
+  , sProgramName="ghc"
+  , sOpt_P_fingerprint=fingerprint0
+  }
+  where
+    platform =
+      Platform{platformWordSize=8
```

Comment:
```
platform = Platform
    { platformWordSize = 8
    , platformOS = OSUnknown
    , platformUnregistered = True
    }
```
That looks a bit more consistent imho and matches what we do elsewhere (and even what you do here in the same file for `Settings`).

Reply:
Note, all this stuff relating to GHC parsing is just for this POC. The IDE already has all this code we wont be replicating it. That said, looks a bit more consistent with what?

Reply:
More consistent with the style we use for formatting records in other places (and the one you use for `Settings` a few lines above).

Reply:
Ah got you now.